src/main/java/software/amazon/cloudformation/proxy/AmazonWebServicesClientProxy.java

```diff
@@ -443,9 +444,24 @@ public Logger getLogger() {
         request.setRequestCredentialsProvider(v1CredentialsProvider);
 
         try {
-            ResultT respose = requestFunction.apply(request);
-            logRequestMetadata(request, respose);
-            return respose;
+            ResultT response = requestFunction.apply(request);
+            logRequestMetadata(request, response);
+            return response;
+        } catch (final Throwable e) {
+            loggerProxy.log(String.format("Failed to execute remote function: {%s}", e.getMessage()));
+            throw e;
+        } finally {
+            request.setRequestCredentialsProvider(null);
+        }
+    }
+
+    public <RequestT extends AmazonWebServiceRequest> void injectCredentialsAndInvoke(final RequestT request,
+                                                                                      final Consumer<RequestT> requestFunction) {
```

Comment:
we should also just make a void function for the v2 proxy as well? if we are tackling the v1 here?

Reply:
I think v2 functions are required to return a result and are never void. @ammokhov can you please confirm?